internal/services/cdn/cdn_frontdoor_security_policy_resource.go

```diff
@@ -273,6 +270,64 @@ func resourceCdnFrontdoorSecurityPolicyRead(d *pluginsdk.ResourceData, meta inte
 	return nil
 }
 
+func resourceCdnFrontdoorSecurityPolicyUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Cdn.FrontDoorSecurityPoliciesClient
+	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	// NOTE: The profile id is used to retrieve properties from the related profile that must match in this security policy
+	profileId, err := profiles.ParseProfileID(d.Get("cdn_frontdoor_profile_id").(string))
+	if err != nil {
+		return err
+	}
+
+	securityPolicyName := d.Get("name").(string)
+	id := securitypolicies.NewSecurityPolicyID(profileId.SubscriptionId, profileId.ResourceGroupName, profileId.ProfileName, securityPolicyName)
+
+	existing, err := client.Get(ctx, id)
+	if err != nil {
+		if !response.WasNotFound(existing.HttpResponse) {
+			return fmt.Errorf("checking for existing %s: %+v", id, err)
+		}
+	}
+
+	profileClient := meta.(*clients.Client).Cdn.FrontDoorProfilesClient
+	resp, err := profileClient.Get(ctx, pointer.From(profileId))
+	if err != nil {
+		return fmt.Errorf("unable to retrieve the 'sku_name' from the CDN FrontDoor Profile(Name: %q)': %+v", profileId.ProfileName, err)
+	}
+
+	profileModel := resp.Model
+
+	if profileModel == nil {
+		return fmt.Errorf("profileModel is 'nil'")
```

Comment:
Will be caught by linter once they have been updated

```suggestion
		return errors.New("profileModel is 'nil'")
```